netty/src/main/java/io/grpc/netty/NettyServer.java

```diff
@@ -78,7 +87,7 @@
   private EventLoopGroup bossGroup;
   private EventLoopGroup workerGroup;
   private ServerListener listener;
-  private Channel channel;
+  private volatile ChannelGroup channelGroup;
```

Comment:
Why is this volatile? It actually looks like it could be `final`

Reply:
channel is added to channelGroup in the eventloop task when bind.

Reply:
Volatile has no effect on that. Volatile only controls the behavior of the variable's direct value. Since the value here never changes, it doesn't need to be volatile.

The issue you mention is impacted by whether ChannelGroup itself is thread-safe or not.

Reply:
It is supposed to be thread-safe and it is explicitly documented, likely we can make it final. (weird that i didn't get consistent result in my debug.)

Reply:
I'm a bit concerned that discussion of "final" seemed to be as a result of thread-safety of ChannelGroup. The two are completely separate; one has no impact on the other.

Reply:
Understood. Just as a result of that fact that there is no `final + volatile`.